enarx-keep/src/backend/probe/x86_64.rs

```diff
@@ -4,19 +4,46 @@ use crate::backend::Datum;
 
 use std::arch::x86_64::{CpuidResult, __cpuid_count};
 use std::convert::Into;
+use std::mem::transmute;
+
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub enum Vendor {
+    Amd,
+    Intel,
+    Unknown,
```

Comment:
Is `Option<Vendor>` better than `Vendor::Unknown`?

Reply:
Yeah, and I think in hindsight I shouldn't have even had `Unknown`. It's probably better for the `Vendor::get()` constructor function to return `Result<Vendor>` and do away with the `Unknown` variant since the `Err` variant of that result is more descriptive.

Reply:
Addressed in latest revision